core/src/HIP/Kokkos_HIP_Parallel_MDRange.hpp

```diff
@@ -227,23 +224,29 @@ class ParallelReduce<FunctorType, Kokkos::MDRangePolicy<Traits...>, ReducerType,
   using DeviceIteratePattern = typename Kokkos::Impl::Reduce::DeviceIterateTile<
       Policy::rank, Policy, FunctorType, WorkTag, reference_type>;
 
-  // Shall we use the shfl based reduction or not (only use it for static sized
-  // types of more than 128bit
-  enum {
-    UseShflReduction = ((sizeof(value_type) > 2 * sizeof(double)) &&
-                        (ValueTraits::StaticValueSize != 0))
-  };
-  // Some crutch to do function overloading
+  // FIXME_HIP_PERFORMANCE Need a rule to choose when to use shared memory and
+  // when to use shuffle
+  static bool constexpr UseShflReduction =
+      (sizeof(value_type) > 2 * sizeof(double)) &&
+      (ValueTraits::StaticValueSize != 0);
+
  private:
-  using DummyShflReductionType  = double;
-  using DummySHMEMReductionType = int;
+  struct ShflReductionTag {};
+  struct SHMEMReductionTag {};
```

Comment:
```suggestion
  struct ShmemReductionTag {};
```
I know it's pre-existent but this capitalization still weird. We can address that in a different pull request, though.

Reply:
What about `ShMemReductionTag` since it's really Shared Memory?